linkerd/router/src/cache.rs

```diff
@@ -121,18 +109,16 @@ where
     ///
     /// Polls the underlying `DelayQueue`. When elements are returned from the
     /// queue, remove the associated key from `values`.
-    fn poll_purge(&mut self) -> Poll<(), ()> {
+    pub fn purge(&mut self) {
```

Comment:
If we made this function always return NotReady (instead of returning nothing), I think it would make it more clear from the API that calling this method will schedule this task to be notified.

Reply:
I think I'd rather rename the function that have a return type of `Poll<Never, Never>` -- it doesn't make the callsite any clearer. Suggestions?

Reply:
Maybe I got the wrong idea about the implicit contract of Poll.  My understanding was that if a method returns NotReady, then the contract is that it *must* schedule the task to be notified.

This method *does* schedule the task to be notified, but that fact isn't communicated by the function signature.  I'm not sure what name would communicate that other than `poll` or `poll_purge`.

Reply:
You are correct that when something returns not ready it must schedule itself to be notified. However, the inverse of this is not true (all functions that schedule notification need not return notready -- especially for crate-internals).